### ISSUE TITLE
WithProxy does not work (as expected)

### ISSUE THREAD

**Issue by chrisbecke** [minio/minio-dotnet#414]
## Issue
I am using minio as a service in a docker stack. As such I have an intenal - insecure - uri that other services use to connect to minio. Additionally there is a reverse proxy that routes external requests to minio.

So, I tried the following example:

```
    // This is the actual address I want to connect to minio over.
    IWebProxy proxy = new WebProxy("minio", 9000);
  
    // files.example.com is the external entrypoint to minio.
    var minio = new MinioClient("files.example.com", "access_key",  "secret_key" )
      .WithProxy(proxy)

    // Now this I am hoping will call minio using the "http://minio:9000" api endpoint
    // but return a pre-signed url using the alternate hostname: "https://files.example.com/bucket/file.txt?....
     String url = await minio.PresignedGetObjectAsync("bucket", "file.txt");
```

But the following gets returned just calling the method:
```
One or more errors occurred. (MinIO API responded with message=The request signature we calculated does not match the signature you provided. Check your key and signing method.)
```
## What I expect
I would expect, with the above example, that Minio would connect to minio:9000 (it does)
Pass "files.example.com" as the http request host header (it does)
Generate the signature using the generated host header. (it does not?)

Clearly the Host header used to generate the signature does not match the generated host header.

## Bonus question
Even if this worked: the generated url would still be http, not https. The scheme is not one of the items signed afaik so I hope here I can simply rewrite the http to https.

**Comment by kannappanr** [comment]
@chrisbecke Can you please paste the output of `mc admin trace -v` when the `pre-signed url` gets executed?

**Comment by chrisbecke** [comment]
Here is the trace from minio:

```
files.example.com [REQUEST s3.ListBuckets] 07:00:02.697
files.example.com GET /
files.example.com Host: files.example.com
files.example.com Content-Length: 0
files.example.com User-Agent: MinIO (Darwin 19.4.0 Darwin Kernel Version 19.4.0: Wed Mar  4 22:28:40 PST 2020; root:xnu-6153.101.6~15/RELEASE_X86_64;X64) minio-dotnet/1.0.9
files.example.com X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
files.example.com X-Amz-Date: 20200518T070001Z
files.example.com Accept: application/json, text/json, text/x-json, text/javascript, application/xml, text/xml
files.example.com Accept-Encoding: gzip, deflate
files.example.com Authorization: AWS4-HMAC-SHA256 Credential=1I2RV7IKX9K790HFGAS9/20200518/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=ba071ac700ed3e8fd2459918673b0db311281d82413e1c9b797bffa349d316c2
files.example.com Connection: Keep-Alive
files.example.com 
files.example.com [RESPONSE] [07:00:02.697] [ Duration 240µs  ↑ 111 B  ↓ 591 B ]
files.example.com 403 Forbidden
files.example.com Vary: Origin
files.example.com X-Amz-Request-Id: 16100D3803E75063
files.example.com X-Xss-Protection: 1; mode=block
files.example.com Accept-Ranges: bytes
files.example.com Content-Length: 334
files.example.com Content-Security-Policy: block-all-mixed-content
files.example.com Content-Type: application/xml
files.example.com Server: MinIO/RELEASE.2020-05-08T02-40-49Z
files.example.com <?xml version="1.0" encoding="UTF-8"?>
<Error><Code>SignatureDoesNotMatch</Code><Message>The request signature we calculated does not match the signature you provided. Check your key and signing method.</Message><Resource>/</Resource><RequestId>16100D3803E75063</RequestId><HostId>1755d2a0-ebc2-4a84-a63d-f73c9d0aa369</HostId></Error>
```

**Comment by sergei66666** [comment]
Hello!
I am not sure, but it can be similar to my solved problem #428 

Try it after release or download master source and use it.

**Comment by harshavardhana** [comment]
Fixed by https://github.com/minio/minio-dotnet/pull/551
